src/core/internalreceiver.cpp

```diff
@@ -0,0 +1,120 @@
+/***************************************************************************
+ internalreceiver.cpp - InternalReceiver
+
+ ---------------------
+ begin                : 22.05.2022
+ copyright            : (C) 2022 by Mathieu Pellerin
+ email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "internalreceiver.h"
+
+#include <QDebug>
+
+InternalReceiver::InternalReceiver( QObject *parent )
+  : AbstractGnssReceiver( parent )
+  , mGeoPositionSource( std::unique_ptr<QGeoPositionInfoSource>( QGeoPositionInfoSource::createDefaultSource( nullptr ) ) )
+{
+  if ( mGeoPositionSource.get() && mGeoPositionSource->error() == QGeoPositionInfoSource::NoError )
+  {
+    mGeoPositionSource->setPreferredPositioningMethods( QGeoPositionInfoSource::AllPositioningMethods );
+    mGeoPositionSource->setUpdateInterval( 1000 );
+
+    connect( mGeoPositionSource.get(), &QGeoPositionInfoSource::positionUpdated, this, &InternalReceiver::handlePositionUpdated );
+    connect( mGeoPositionSource.get(), qOverload<QGeoPositionInfoSource::Error>( &QGeoPositionInfoSource::error ), this, &InternalReceiver::handleError );
+
+    setValid( true );
+  }
+}
+
+void InternalReceiver::handleDisconnectDevice()
+{
+  mGeoPositionSource->stopUpdates();
+  mLastGnssPositionValid = false;
+}
+
+void InternalReceiver::handleConnectDevice()
+{
+  mGeoPositionSource->startUpdates();
+}
+
+void InternalReceiver::handlePositionUpdated( const QGeoPositionInfo &positionInfo )
+{
+  if ( mLastGnssPositionValid && !positionInfo.coordinate().isValid() )
+  {
+    return;
```

Comment:
Do we also propagate things like position lost?

Reply:
@m-kuhn , no, the established behavior is for the last valid position to "stick" until a new valid position is provided. In the positioning information panel, if the last valid position is too old (30sec I think), we then display the information in a gray colored font.